site/components/map/ServicesMap.tsx

```diff
@@ -222,120 +224,129 @@ const Map = ({
         setPopupInfo({});
     }, []);
 
-    const selectAllStops = async (evt: SyntheticEvent) => {
-        evt.preventDefault();
+    const addSelectedStopsAndServices = async (includeMarkerData?: boolean) => {
+        const parsed = z.array(stopSchema).safeParse(searched);
+        if (!parsed.success) {
+            stateUpdater({
+                ...state,
+                errors: [
+                    ...state.errors.filter((err) => !Object.keys(servicesConsequenceSchema.shape).includes(err.id)),
+                    ...flattenZodErrors(parsed.error),
+                ],
+            });
+        } else {
+            if (showSelectAllText) {
+                const atcoCodes = includeMarkerData ? markerData.map((marker) => marker.atcoCode) : [];
+
+                const servicesInPolygon = includeMarkerData
+                    ? dataSource
+                        ? await fetchServicesByStops({ atcoCodes, includeRoutes: true, dataSource: dataSource })
+                        : await fetchServicesByStops({ atcoCodes, includeRoutes: true })
+                    : [];
+
+                const servicesStopsInPolygon = servicesInPolygon.flatMap((service) => service.stops);
+                const markerDataInAService = includeMarkerData
+                    ? markerData
+                          .filter((marker) => servicesStopsInPolygon.includes(marker.atcoCode))
+                          .map((marker) => {
+                              const services = servicesInPolygon.filter((service) =>
+                                  service.stops.includes(marker.atcoCode),
+                              );
+                              return {
+                                  ...marker,
+                                  serviceIds: services.length > 0 ? services.map((s) => s.id) : undefined,
+                              };
+                          })
+                    : [];
+
+                const servicesToAdd = servicesInPolygon
+                    .filter((service) =>
+                        service.stops.filter((stop) => {
```

Comment:
because .includes returns a true or false value we don't need the if statement

`service.stops.filter((stop) => markerData.map((marker) => marker.atcoCode).includes(stop))`